src/org/mozilla/javascript/MemberBoxSlot.java

```diff
@@ -0,0 +1,120 @@
+        // It sounds logical that this would be the same as the logic for a normal Slot,
+        // but the spec is super pedantic about things like the order of properties here,
+        // so we need special support here.
+        ScriptableObject desc = (ScriptableObject) cx.newObject(scope);
+        String fName = name == null ? "f" : name.toString();
+        desc.defineProperty(
+                "enumerable",
+                (getAttributes() & ScriptableObject.DONTENUM) == 0,
+                ScriptableObject.EMPTY);
+        desc.defineProperty(
+                "configurable",
+                (getAttributes() & ScriptableObject.PERMANENT) == 0,
+                ScriptableObject.EMPTY);
+        if (getter == null && setter == null) {
+            desc.defineProperty(
+                    "writable",
+                    (getAttributes() & ScriptableObject.READONLY) == 0,
+                    ScriptableObject.EMPTY);
+        }
+
+        if (getter != null) {
+            desc.defineProperty(
+                    "get", getter.asGetterFunction(fName, scope), ScriptableObject.EMPTY);
+        }
+        if (setter != null) {
+            desc.defineProperty(
+                    "set", setter.asSetterFunction(fName, scope), ScriptableObject.EMPTY);
+        }
+        return desc;
+    }
+
+    @Override
+    public boolean setValue(Object value, Scriptable owner, Scriptable start) {
+        if (setter == null) {
+            if (getter != null) {
+                throwNoSetterException(start, value);
+                return true;
+            }
+        } else {
+            Context cx = Context.getContext();
+            Class<?>[] pTypes = setter.argTypes;
+            // XXX: cache tag since it is already calculated in
+            // defineProperty ?
+            Class<?> valueType = pTypes[pTypes.length - 1];
+            int tag = FunctionObject.getTypeTag(valueType);
+            Object actualArg = FunctionObject.convertArg(cx, start, value, tag);
+            Object setterThis;
+            Object[] args;
+            if (setter.delegateTo == null) {
+                setterThis = start;
+                args = new Object[] {actualArg};
+            } else {
+                setterThis = setter.delegateTo;
+                args = new Object[] {start, actualArg};
+            }
+            setter.invoke(setterThis, args);
+            return true;
+        }
+        return super.setValue(value, owner, start);
+    }
+
+    @Override
+    public Object getValue(Scriptable start) {
+        if (getter != null) {
+            Object getterThis;
+            Object[] args;
+            if (getter.delegateTo == null) {
+                getterThis = start;
+                args = ScriptRuntime.emptyArgs;
+            } else {
+                getterThis = getter.delegateTo;
+                args = new Object[] {start};
+            }
+            return getter.invoke(getterThis, args);
```

Comment:
Maybe we can let this look more simple

    if (getter.delegateTo == null) {
        return getter.invoke(start, ScriptRuntime.emptyArgs);
    }
    return getter.invoke(getter.delegateTo, new Object[] {start});

Reply:
Yeah, I cut and pasted and I don't know why that code was so weird.